spec/features/admin/sign_in_as_principal_spec.rb

```diff
@@ -0,0 +1,46 @@
+RSpec.feature 'Move advisers between firms' do
+  let(:admin_firm_page) { Admin::FirmPage.new }
+  let(:firms_index_page) { SelfService::FirmsIndexPage.new }
+  let(:updated_postcode) { 'EH11 2AB' }
+
+  scenario 'admin signs in as principal' do
+    given_there_is_a_fully_registered_principal_user
+    and_they_have_a_firm_with_an_adviser
+    and_i_am_logged_in_as_an_admin_user
+    and_i_am_on_the_admin_firm_page
+    when_i_sign_in_as_principal
+    then_i_see "You are now signed in as #{@principal.full_name}"
+    and_i_am_on_the_self_service_firm_page
+  end
+
+  def given_there_is_a_fully_registered_principal_user
+    @principal = FactoryGirl.create(:principal)
+    @user = FactoryGirl.create(:user, principal: @principal)
+  end
+
+  def and_they_have_a_firm_with_an_adviser
+    @adviser = FactoryGirl.create :adviser, postcode: updated_postcode
+    firm_attrs = FactoryGirl.attributes_for(:firm, fca_number: @principal.fca_number)
+    @principal.firm.advisers << @adviser
+    @principal.firm.update_attributes(firm_attrs)
+  end
+
+  def and_i_am_logged_in_as_an_admin_user
+  end
+
+  def and_i_am_on_the_admin_firm_page
+    admin_firm_page.load(firm_id: @principal.firm.id)
+  end
+
+  def when_i_sign_in_as_principal
+    admin_firm_page.sign_in_as_principal.click
+  end
+
+  def then_i_see(message)
+    expect(page.text).to include(message)
+  end
+
+  def and_i_am_on_the_self_service_firm_page
+    expect(firms_index_page).to be_displayed
+  end
```

Comment:
Also, got a failin test for ya: 

```ruby
  scenario 'admin signs in as a principal without a user record' do
    # i.e. when the invite rake task hasn't been run
    given_there_is_a_principal
    and_i_am_logged_in_as_an_admin_user
    and_i_am_on_the_admin_firm_page
    then_it_doesnt_blow_up
  end

  def given_there_is_a_principal
    @principal = FactoryGirl.create(:principal)
  end

  def then_it_doesnt_blow_up
    expect(admin_firm_page).to_not have_text('NoMethodError')
  end
```

Suggest maybe hiding it for those principals — since it shouldn't occur in normal usage after we've run the rake task but it also probably shouldn't 500